includes/class-wc-stripe-intent-controller.php

```diff
@@ -414,6 +414,18 @@ public function update_payment_intent( $payment_intent_id = '', $order_id = null
 			if ( '' !== $selected_upe_payment_type ) {
 				// Only update the payment_method_types if we have a reference to the payment type the customer selected.
 				$request['payment_method_types'] = [ $selected_upe_payment_type ];
+				if ( WC_Stripe_UPE_Payment_Method_CC::STRIPE_ID === $selected_upe_payment_type ) {
+					if ( in_array(
+						WC_Stripe_UPE_Payment_Method_Link::STRIPE_ID,
+						$gateway->get_upe_enabled_payment_method_ids(),
+						true
+					) ) {
```

Comment:
Nit: there is no separate logic between both if statements, so I'd do all checks in just one `if`.

Reply:
changed it. thank you!